docs/how-tos/upload-data.md

```diff
@@ -0,0 +1,238 @@
+
+```python {class="line-numbers linkable-line-numbers"}
+import os  # Add this package at the top of your program
+# with your other imports
+
+my_data_directory = "/Users/Artoo/my_cat_photos"
+
+for file_name in os.listdir(my_data_directory):
+    await data_client.file_upload_from_path(
+      part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+      tags=["cat", "animals", "brown"],
+      filepath=os.path.join(my_data_directory, file_name)
+    )
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% /tablestep %}}
+{{% tablestep %}}
+{{<imgproc src="/services/icons/data-cloud-sync.svg" class="fill alignleft" style="max-width: 150px" declaredimensions=true alt="Cloud connection">}}
+**3. Run your code**
+
+Save and run your code once.
+Running your code more than once will duplicate the data.
+View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+{{% /tablestep %}}
+{{< /table >}}
+
+## Upload images with the Viam mobile app
+
+Upload images as machine data straight from your phone, skipping the normal data capture and cloud synchronization process, through the [Viam mobile app](/fleet/control/#control-interface-in-the-viam-mobile-app).
+This is useful if you want to capture images for training machine learning models on the go.
+
+### Prerequisites
+
+{{< expand "Download the Viam mobile app and sign into your Viam account" >}}
+
+Install the mobile app from the [App Store](https://apps.apple.com/vn/app/viam-robotics/id6451424162) or [Google Play](https://play.google.com/store/apps/details?id=com.viam.viammobile&hl=en&gl=US).
+
+<a href="https://apps.apple.com/vn/app/viam-robotics/id6451424162" target="_blank">
+  <img src="https://github.com/viamrobotics/docs/assets/90707162/a470b65d-1b97-412f-9f97-daf902f2f053" width="200px" alt="apple store icon" class="center-if-small" >
+</a>
+
+<a href="https://play.google.com/store/apps/details?id=com.viam.viammobile&hl=en&gl=US" target="_blank">
+  <img src="https://github.com/viamrobotics/docs/assets/90707162/6ebd6960-08c5-41d4-81f9-42293fbfdfd4" width="200px" alt="google play store icon" class="center-if-small" >
+</a>
+
+{{< /expand >}}
+
+### Instructions
+
+{{< table >}}
+{{% tablestep link="/services/data/" %}}
+**1. Navigate to your machine**
+
+Select an organization by clicking on the menu icon in the top left corner and tapping an organization.
+
+Tap the **Locations** tab and select a location, then select the machine you want your data to be associated with.
+
+{{% /tablestep %}}
+{{% tablestep %}}
+**2. Upload images**
+
+Tap the menu button marked "**...**" in the upper right corner.
+Tap **Upload Images**.
+
+Select each image you want to upload, then tap **Add**.
+
+The uploaded images metadata will contain the machine part you selected.
+However, the uploaded images will not be associated with a component or method.
+
+{{% /tablestep %}}
+{{< /table >}}
```

Comment:
Add a next steps section please. I think these may be useful:
- https://docs-test.viam.dev/3367/how-tos/image-data/
- https://docs-test.viam.dev/3367/how-tos/deploy-ml/